messagebird/http_client.py

```diff
@@ -29,21 +34,22 @@ def request(self, path, method='GET', params=None):
             'Content-Type': 'application/json'
         }
 
-        if method == 'DELETE':
-            response = requests.delete(url, verify=True, headers=headers, data=json.dumps(params))
-        elif method == 'GET':
-            response = requests.get(url, verify=True, headers=headers, params=params)
-        elif method == 'PATCH':
-            response = requests.patch(url, verify=True, headers=headers, data=json.dumps(params))
-        elif method == 'POST':
-            response = requests.post(url, verify=True, headers=headers, data=json.dumps(params))
-        elif method == 'PUT':
-            response = requests.put(url, verify=True, headers=headers, data=json.dumps(params))
-        else:
-            raise ValueError(str(method) + ' is not a supported HTTP method')
+        method_switcher = {
+            'DELETE': requests.delete(url, verify=True, headers=headers, data=json.dumps(params)),
+            'GET': requests.get(url, verify=True, headers=headers, params=params),
+            'PATCH': requests.patch(url, verify=True, headers=headers, data=json.dumps(params)),
+            'POST': requests.post(url, verify=True, headers=headers, data=json.dumps(params)),
+            'PUT': requests.put(url, verify=True, headers=headers, data=json.dumps(params))
+        }
+        response = method_switcher.get(method, str(method) + ' is not a supported HTTP method')
+        if isinstance(response, str):
+            raise ValueError(response)
 
         if response.status_code in self.__supported_status_codes:
-            response_text = response.text
+            if format == ResponseFormat.text:
```

Comment:
Use something like you did for `method_switcher` please. And return a `response` not a `response_text`.